### DISCUSSION TITLE
Pocketbase JS-SDK subscribe "*" does not fire when view requirement is false.

### DISCUSSION THREAD

**Discussion by jonshipman** [pocketbase/pocketbase#4080] [Q&A] [answered]
Scenario: A collection with a boolean field "active" that has a view + search requirement `@request.auth.id != "" && active = true`.

`pb.collection('collection').subscribe((e)=>{console.log(e);});` does not fire when active is switched to false. Happens with single or multiple records.

In my head, I'd assume that a subscribe would return an empty set for updates that don't meet the current view requirements. Unknown how to solve other than handling the permissions in the client, which wouldn't be as secure as denying access in PB.

**Comment by ganigeorgiev** [comment, marked as answer]
This is working as intended.

The hook is not firing because we can't be sure (at least I couldn't find an easy and performant way to do it) whether the client had previously access to the old record state (or even whether they previously fetched the record in the first place so that they need the notification for the lack of permissions to the new state).

> Unknown how to solve other than handling the permissions in the client, which wouldn't be as secure as denying access in PB.

It is not clear in what context do you use the subscribe and why this is a security problem.
Technically your client would have the old/"cached" record state that they previously had permissions to.
On refresh or once they send a new view request with the specific record they will get an error response.
I could understand from an UX point that it is a problem and I have it in my never ending todo to revisit this sometime in the future because there were also several requests for sending the changed/old fields alongside the new record state, which has the same limitation as above, but it is left aside for now.

**Reply by jonshipman** [reply]
By "secure" I mean a solution could be that I remove active = true from the view requirements. The reprecussions would be that active = false would be visible via the Pocketbase API and the filtering would need to be done client side. Just mostly musing to myself. The situation is a WordPress plugin that is sending data to a Pocketbase database. On plugin activation the site creates an entry in a table and on deactivation 'active' is set to false.
